programs/pkey/pk_sign.c

```diff
@@ -70,6 +73,13 @@ int main( int argc, char *argv[] )
     const char *pers = "mbedtls_pk_sign";
     size_t olen = 0;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platofrm_setup returned %d\n", ret );
+        goto exit;
```

Comment:
This should be changed to `return ret`. If you go to exit here, you will free uninitialized resources, which might lead to undefined behavior, depending on implementation.